backend/plan/views.py

```diff
@@ -399,3 +406,101 @@ def get_queryset(self):
             "sections__meetings__room",
         )
         return queryset
+
+
+class CalendarAPIView(APIView):
+    schema = PcxAutoSchema(
+        response_codes={
+            reverse_func("calendar-view", args=["schedule_pk"]): {
+                "GET": {
+                    200: "Schedule exported successfully",
+                },
+            },
+        },
+    )
+
+    def get(self, *args, **kwargs):
+        """
+        Return a .ics file of the user's selected schedule
+        ---
+        responses:
+            "200":
+                description: Return a calendar file in ICS format.
+                content:
+                    text/calendar:
+                        schema:
+                            type: string
+        ---
+        """
+        schedule_pk = kwargs["schedule_pk"]
+
+        schedule = (
+            Schedule.objects.filter(pk=schedule_pk)
+            .prefetch_related("sections", "sections__meetings")
+            .first()
+        )
+
+        if not schedule:
+            return Response({"detail": "Invalid schedule"}, status=status.HTTP_403_FORBIDDEN)
+
+        day_mapping = {"M": "MO", "T": "TU", "W": "WE", "R": "TH", "F": "FR"}
+
+        calendar = ICSCal(creator="Penn Labs")
+        calendar.extra.append(ContentLine(name="X-WR-CALNAME", value=f"{schedule.name} Schedule"))
+
+        for section in schedule.sections.all():
+            e = ICSEvent()
+            e.name = section.full_code
+            e.created = timezone.now()
+
+            days = []
+            for meeting in section.meetings.all():
+                days.append(day_mapping[meeting.day])
+            first_meeting = list(section.meetings.all())[0]
+
+            start_time = str(Meeting.int_to_time(first_meeting.start))
```

Comment:
Just a general question -- is there a reason we do this from the backend instead of the frontent?

Reply:
When I first built the feature I had all the logic done on the frontend, but Rohan suggested that logic be moved to the backend to simplify the frontend code and make it look more visually appealing

Reply:
Ok yeah I think the link is maybe more intuitive.